gems/pending/openstack/amqp/openstack_ceilometer_event_monitor.rb

```diff
@@ -0,0 +1,76 @@
+require 'openstack/openstack_event_monitor'
+require 'openstack/amqp/openstack_amqp_event'
+require 'openstack/amqp/openstack_ceilometer_event_converter'
+
+class OpenstackCeilometerEventMonitor < OpenstackEventMonitor
+  def self.available?(_options = {})
+    # TODO(maufart): add lookup for ceilometer_enabled in vmdb config (from MiqRunner?)
+    true
+  end
+
+  def self.plugin_priority
+    1
+  end
+
+  def initialize(options = {})
+    @options = options
+    @ems = options[:ems]
+  end
+
+  def start
+    @since          = nil
+    @monitor_events = true
+  end
+
+  def stop
+    @monitor_events = false
+  end
+
+  def provider_connection
+    @provider_connection ||= @ems.connect(:service => "Metering")
+  end
+
+  def each_batch
+    while @monitor_events
+      puts ""
+      puts "####################################### quering ceilometer for newer than #{latest_event_timestamp}"
+      puts ""
+      query_options = [{
+          'field' => 'start_timestamp',
+          'op' => 'gt',
+          'value' => latest_event_timestamp || ''
+      }]
+      # TODO(maufart): add filtered events fetch method into fog-openstack and refactor then
+      events = list_events(query_options).sort_by { |e| e.generated }
+      @since = events.last.generated unless events.empty?
+
+      amqp_events = events.map do |event|
+        converted_event = OpenstackCeilometerEventConverter.new(event)
+        puts "---------AMQP ceilometer-----------"
+        puts "event:"
+        pp event
+        amqp_event(nil, converted_event.metadata, converted_event.payload)
+      end
+
+      yield amqp_events
+    end
+  end
+
+  def each
+    each_batch do |events|
+      events.each { |e| yield e }
+    end
+  end
+
+  private
+
+  def list_events(query_options)
```

Comment:
@tzumainn This is a workaround because current fog-openstack does not support filtering events. Then events(options) would be used instead of list_events(options) and creating Fog Event objects.